pallets/marketplace/src/lib.rs

```diff
@@ -0,0 +1,486 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![allow(clippy::upper_case_acronyms)]
+
+use frame_support::{
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency, ExistenceRequirement, NamedReservableCurrency, OriginTrait},
+	transactional,
+};
+use frame_system::{ensure_signed, RawOrigin};
+use sp_runtime::{
+	traits::{Saturating, StaticLookup},
+	Percent,
+};
+
+use types::{Offer, TokenInfo};
+use weights::WeightInfo;
+
+use pallet_nft::{types::ClassType, Instances};
+use primitives::ReserveIdentifier;
+
+mod benchmarking;
+pub mod types;
+pub mod weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub type BalanceOf<T> =
+	<<T as pallet_nft::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type TokenInfoOf<T> = TokenInfo<BalanceOf<T>>;
+pub type OfferOf<T> =
+	Offer<<T as frame_system::Config>::AccountId, BalanceOf<T>, <T as frame_system::Config>::BlockNumber>;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::OriginFor;
+
+	pub const RESERVE_ID: ReserveIdentifier = ReserveIdentifier::Marketplace;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn tokens)]
+	/// Stores token info
+	pub type Tokens<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, T::NftClassId, Twox64Concat, T::NftInstanceId, TokenInfoOf<T>, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn offers)]
+	/// Stores offer info
+	pub type Offers<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		(T::NftClassId, T::NftInstanceId),
+		Twox64Concat,
+		T::AccountId,
+		OfferOf<T>,
+		OptionQuery,
+	>;
+
+	/// Offer existence check by account and token id
+	#[pallet::storage]
+	#[pallet::getter(fn offers_by_maker)]
+	pub type OffersByMaker<T: Config> = StorageNMap<
```

Comment:
I wonder why this cannot be combined into one storage together with the `Offers`.

If you make order of keys like `(NftClassId, NftInstanceId, AccountId)` and make the Offer type as storage item -> you actually get the same result .

Reply:
true, initially i thought i would need remove offers by account, but i only need to remove offers by token id